modules/concom/vue/components/StaffSidebarEmail.js

```diff
@@ -31,6 +33,21 @@ function onSave() {
   this.$emit('saveEmailClicked', data);
 }
 
+async function onDelete() {
+  try {
+    const confirmBoxMessage = `Really delete the e-mail address "${this.email}"?`;
+    await confirmbox(confirmBoxTitle, confirmBoxMessage);
```

Comment:
Short answer is: Yes, it will handle this fine.

Longer answer with some context:

Javascript has gone through a couple of iterations on how it handles asynchronous code. In the beginning, it was just callbacks, but introduced callback hell.

Eventually, they added the Promise API. Promises allowed us to handle asynchronous code in a way that was a little cleaner than just callbacks. You have the Promise, and then `.then()` or `.catch()` to handle what happens if the function was successful or failed. This is good, but over time people realized that it was very verbose and could be difficult to read if you had to chain lots of promises. What if, in response to a promise, you need to execute another promise? Well, you need another `.then()` or `.catch()` call. You can chain them, but that can get messy.

Enter `async` and `await`. These are just another way to execute asynchronous code, and allow us to execute multiple asynchronous calls more easily with a syntax that is easier to read. You can use `try/catch` to handle error cases. Typically, I think anything that returns a Promise can be `await`ed, and when the promise resolves, it will proceed with the next call, and when it is rejected it will be sent to the catch block. I don't think you can make a promise out of an async function call, however.

The code above is roughly equivalent to this, if we went the promise route:

```
return new Promise((resolve, reject) => {
  const confirmBoxMessage = `Really delete the e-mail address "${this.email}"?`;
  return confirmBox(confirmBoxTitle, confirmBoxMessage).then(() => {
    const data = {
      id: this.emailId
    };

    this.$emit('deleteEmailClicked', data);
  }).catch((error) => {
    // User cancelled
  });
});
```